python/ray/data/tests/test_dataset.py

```diff
@@ -329,7 +329,7 @@ def test_batch_tensors(ray_start_regular_shared):
     with pytest.raises(pa.lib.ArrowInvalid):
```

Comment:
Can we add tests that this is working properly with the flag on and off?
- Test that after `map_batches()` with a UDF returning a pandas DF the _dataset_format is "pandas".
- Test that after `from_pandas()` the _dataset_format is "pandas".

Also test that with the flag off, the format is "arrow".

Reply:
@ericl I've updated the test code. Could you review it again?